qiskit_experiments/library/randomized_benchmarking/rb_experiment.py

```diff
@@ -214,25 +224,68 @@ def _generate_circuit(
+
+        return options
+
+    def _finalize(self):
+        super()._finalize()
+
+        # Set constraints to compute basis gates EPGs from an estimated EPC
+        gate_error_ratio = self.experiment_options.gate_error_ratio
+
+        if gate_error_ratio == "default":
+            num_qubit_default_basis = {
+                1: {"sx": 1.0, "rz": 0.0, "x": 1.0},
+                2: {"cx": 1.0},
+            }
+            qubits = tuple(self.physical_qubits)
+            default_ratio = num_qubit_default_basis.get(len(qubits), None)
+            if default_ratio:
+                gate_error_ratio = {(qubits, k): v for k, v in default_ratio.items()}
+            else:
+                gate_error_ratio = None
+
+        if gate_error_ratio is not None:
+            self.analysis.set_options(gate_error_ratio=gate_error_ratio)
+
+            gates_in_estimates = set(
+                q_gate_tup[1] for q_gate_tup in gate_error_ratio.keys()
+            )
+            # Validate if assumed gates are a part to basis gates in the transpile options
+            basis_gates = set(self.transpile_options.basis_gates)
+            if not basis_gates.issuperset(gates_in_estimates):
+                raise ValueError(
+                    f"Assumed gates {gates_in_estimates} is not valid subset of basis gates {basis_gates}."
+                )
 
     def _transpiled_circuits(self) -> List[QuantumCircuit]:
         """Return a list of experiment circuits, transpiled."""
         transpiled = super()._transpiled_circuits()
-        for c in transpiled:
-            meta = self._get_circuit_metadata(c)
-            if meta is not None:
-                c_count_ops = RBUtils.count_ops(c, self.physical_qubits)
-                circuit_length = meta["xval"]
-                count_ops = [(key, (value, circuit_length)) for key, value in c_count_ops.items()]
-                meta.update({"count_ops": count_ops})
+
+        if self.analysis.options.gate_per_clifford is not None or self.analysis.options.gate_error_ratio is None:
+            # This number is pre-computed by user in some other way or gate errors are not computed.
+            return transpiled
+
+        # Compute average basis gate numbers per Clifford operation
+        # This is probably main source of performance regression.
+        # This should be integrated into transpile pass in future.
+        gate_per_clifford = defaultdict(int)
+        total_cliffs = np.sum(self.experiment_options.lengths)
+        for circ in transpiled:
+            for (qubits, instr), count in RBUtils.count_ops(circ, self.physical_qubits).items():
+                if instr == "measure":
+                    continue
+                # This is qubit aware count opts
+                gate_per_clifford[(qubits, instr)] += count / total_cliffs
+
+        # Directly update analysis option
+        self.analysis.set_options(gate_per_clifford=gate_per_clifford)
```

Comment:
Can we upgrade base experiments so that this value can be passed to experiment data metadata? Otherwise EPGs will disappear in analysis results when we rerun analysis with retrieved experiment data.

Reply:
This is resolved by swapping the order of transpile and creation of experiment data in the base experiment.